app/lib/ability.rb

```diff
@@ -7,6 +7,15 @@ def initialize(user)
       return
     end
 
+    # Custom actions
+    alias_action :flag, :toggle_field, :edit_take_action, :update_take_action,
+                 :unlock, :edit_13614c_form_page1, :edit_13614c_form_page2,
+                 :edit_13614c_form_page3, :save_and_maybe_exit,
+                 :update_13614c_form_page1, :update_13614c_form_page2,
+                 :update_13614c_form_page3, :cancel_13614c,
+                 :resource_to_client_redirect,
+                 to: :hub_client_management
+
```

Comment:
I think there are two kinds of actions in here. Some, like `flag`, `unlock`, and `toggle_field` probably make since to be real permissions. Others, like `edit_13614c_form_page1` feel like they should probably be their own sub resource of client (even if it's just a PORO with `ActiveModel` included). In this way, you can tidy the controller and the permissions and use the authorize parent resource functionality of CanCanCan.

I didn't go that far because it's a significant refactor and I felt like this was a good first step anyways.